components/org.wso2.bfsi.consent.management.common/src/main/java/org/wso2/bfsi/consent/management/common/persistence/JDBCPersistenceManager.java

```diff
@@ -0,0 +1,154 @@
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+/**
+ * This class is used for handling consent management data persistence in the JDBC Store. During the server
+ * start-up, it checks whether the database is created, if not it creates one. It reads the data source properties
+ * from the bfsi-consent-management.xml. This is implemented as a singleton. An instance of this class can be obtained
+ * through JDBCPersistenceManager.getInstance() method.
+ */
+public class JDBCPersistenceManager {
+
+    private static volatile JDBCPersistenceManager instance;
+    private static volatile DataSource dataSource;
+    private static final Log log = LogFactory.getLog(JDBCPersistenceManager.class);
+
+    private JDBCPersistenceManager() {
+
+        initDataSource();
+    }
+
+    /**
+     * Get an instance of the JDBCPersistenceManager. It implements a double checked locking initialization.
+     *
+     * @return JDBCPersistenceManager instance
+     */
+    public static synchronized JDBCPersistenceManager getInstance() {
+        if (instance == null) {
+            synchronized (JDBCPersistenceManager.class) {
+                if (instance == null) {
+                    instance = new JDBCPersistenceManager();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Initialize the data source.
+     */
+    private void initDataSource() {
+
+        if (dataSource != null) {
+            return;
+        }
+        synchronized (JDBCPersistenceManager.class) {
+            try {
+                String dataSourceName = ConsentManagementConfigParser.getInstance().getDataSourceName();
+                if (StringUtils.isNotBlank(dataSourceName)) {
+                    Context context = new InitialContext();
+                    dataSource = (DataSource) context.lookup(dataSourceName);
+                } else {
+                    throw new ConsentManagementRuntimeException("Persistence Manager configuration for Consent " +
+                            "Management is not available in bfsi-consent-management.xml file. " +
+                            "Terminating the JDBC persistence manager initialization.");
+                }
+            } catch (NamingException e) {
+                throw new ConsentManagementRuntimeException("Error when looking up the Consent Management Data Source.",
+                        e);
+            }
+        }
+    }
+
+    /**
+     * Returns an database connection for Consent Management data source.
+     *
+     * @return Database connection.
+     * @throws ConsentManagementRuntimeException Exception occurred when getting the data source.
+     */
+    public Connection getDBConnection() throws ConsentManagementRuntimeException {
+
+        try {
+            Connection dbConnection = dataSource.getConnection();
+            dbConnection.setAutoCommit(false);
```

Comment:
does this override the auto commit configuration configured in the deployment.toml file?

Reply:
This will be a separate connector on top IS. This will not override any data existing data source configs. hence we will be manually creating the data source when adding the connector. We are manually committing all the transactions from DAO hence kept this as false.